setup.py

```diff
@@ -50,13 +50,14 @@
         'Programming Language :: Python :: 3.6',
         'Programming Language :: Python :: 3.7',
         'Programming Language :: Python :: 3.8',
+        'Programming Language :: Python :: 3.9',
         'Topic :: Internet :: WWW/HTTP',
         'Topic :: Internet :: WWW/HTTP :: WSGI',
         'Topic :: Internet :: WWW/HTTP :: WSGI :: Middleware',
         'Topic :: Text Processing :: Markup :: XML',
     ],
     install_requires=[
-        'setuptools',
+    #    'setuptools',
```

Comment:
with setuptools in install_requires, it did not work on python 2.7 for me. That is because of a bug in tox4 or so, that it duplicate a requirement to install, as setuptools and setuptools 40.8.0 should be installed.

From a pip perspective that did not make sense. Especially that should not break the build. 

As setuptools is a implicite dependency it still works.

Reply:
oh and by the way, from my point of view your assumption at https://github.com/jugmac00/til/blob/master/zope/why-is-setuptools-a-runtime-requirement-for-zope-projects.md is wrong. The requirement of setuptools due to the namespace packaging is wrong and obsolte. First of all setup.py depends itself on setuptools which makes this a recursive dependency, and all current versions of python (including 2.7) brings an internal setuptools version that is new enough to have the call for declaring namespaces.

Also and more important:
```
>>> import this
The Zen of Python, by Tim Peters

Beautiful is better than ugly.
Explicit is better than implicit.
Simple is better than complex.
Complex is better than complicated.
Flat is better than nested.
Sparse is better than dense.
Readability counts.
Special cases aren't special enough to break the rules.
Although practicality beats purity.
Errors should never pass silently.
Unless explicitly silenced.
In the face of ambiguity, refuse the temptation to guess.
There should be one-- and preferably only one --obvious way to do it.
Although that way may not be obvious at first unless you're Dutch.
Now is better than never.
Although never is often better than *right* now.
If the implementation is hard to explain, it's a bad idea.
If the implementation is easy to explain, it may be a good idea.
Namespaces are one honking great idea -- let's do more of those!
```
The last line. Namespaces are a very good idea and should be used more often and not avoid like your text implies.

Reply:
About namespaces... maybe I need to redo my text.

There was no intention to suggest you should avoid namespaces in general, but only `pkg_resource-style namespace packages`.

The official Python 3 documentation recommends to use `native namespace packages` when you need them for a new package.
https://packaging.python.org/guides/packaging-namespace-packages/#native-namespace-packages